### PR TITLE
save protobuf generic schema as Avro metadata #265

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: scio-core/src/main/java/com/google/cloud/dataflow/sdk/io/PatchedAvroIO.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
```
Comment (ravwojdyla) [inline scio-core/src/main/java/com/google/cloud/dataflow/sdk/io/PatchedAvroIO.java]: I don't really like how to is diverging from SDK/Beam. We essentially don't want to have this file. Could you create a PR/issue for beam to see if they would accept this?

**Comment by ravwojdyla** [comment]
Is this feature really that important? Is dump raw not enough? Is the use case this is solving just eye balling the results? If one wants to process those PB, they should use compiled protobuf and do it the right way.

File: scio-core/src/main/java/com/google/cloud/dataflow/sdk/io/PatchedAvroIO.java (hunk 1 above)
Comment (nevillelyh) [inline scio-core/src/main/java/com/google/cloud/dataflow/sdk/io/PatchedAvroIO.java]: Yes but it'll take a while for them to backport to DataflowJavaSDK so this file will exist for a while.

Comment (nevillelyh) [inline scio-core/src/main/java/com/google/cloud/dataflow/sdk/io/PatchedAvroIO.java]: Sure will do

**Comment by nevillelyh** [comment]
@ravwojdyla this is synced with upstream PR https://github.com/apache/incubator-beam/pull/1177 and ready to merge. It writes output file readable with https://github.com/spotify/gcs-tools/pull/1/

**Pull request merged**